xncml/core.py

```diff
@@ -278,3 +286,127 @@ def to_ncml(self, path=None):
         xml_output = xmltodict.unparse(self.ncroot, pretty=True)
         with open(path, 'w') as fd:
             fd.write(xml_output)
+
+    def to_cf_dict(self):
+        """Convert internal representation to a CF-JSON dictionary.
+
+        The CF-JSON specification includes `data` for variables, but this is not included here.
+
+        Returns
+        -------
+        Dictionary with `dimensions` and `variables` keys. May also optionally include an `attributes` key and a
+        `groups` key. Additional keys prefixed with `@` may be included for <netcdf> tag attributes,
+        for example `@location`.
+
+        References
+        ----------
+        http://cf-json.org/specification
+        """
+        res = OrderedDict()
+        nc = self.ncroot['netcdf']
+
+        for key, val in nc.items():
+            if key[0] == '@':
+                res[key] = val
+            if key == 'dimension':
+                res.update(_dims_to_json(val))
+            if key == 'group':
+                res.update(_groups_to_json(val))
+            if key == 'attribute':
+                res.update(_attributes_to_json(val))
+            if key == 'variable':
+                res.update(_variables_to_json(val))
+
+        return res
+
+
+def _dims_to_json(dims: list) -> dict:
+    """The dimensions object has dimension id:size as its key:value members."""
+    out = OrderedDict()
+    for dim in dims:
+        if int(dim['@length']) > 1:
+            out[dim['@name']] = int(dim['@length'])
+
+    return {'dimensions': out}
+
+
+def _groups_to_json(groups: list) -> dict:
+    out = OrderedDict()
+    for group in groups:
+        name = group['@name']
+        out[name] = OrderedDict()
+        if 'attribute' in group:
+            out[name].update(_attributes_to_json(group['attribute']))
+        if 'group' in group:
+            out[name].update(_groups_to_json(group['group']))
+
+    return {'groups': out}
+
+
+def _attributes_to_json(attrs: list) -> dict:
+    """The attributes object contains arbitrary attributes as its key:value members."""
+    SPECIAL_ATTRS = ['missing_value', 'cell_methods']
+
+    out = OrderedDict()
+    for attr in attrs:
+        if attr['@name'] not in SPECIAL_ATTRS:
+            try:
+                out[attr['@name']] = _cast(attr)
+            except ValueError as exc:
+                warn(f"Could not cast {attr['@name']}:\n{exc}")
+
+    return {'attributes': out}
+
+
+def _variables_to_json(variables: list) -> dict:
+    """The variables definition object has variable id:object as its key:value members.
+
+    Each variable object MUST include shape, attributes and data objects.
+    The shape field is an array of dimension IDs which correspond to the array ordering of the variable data.
+    """
+    AXIS_VAR = ['time', 'lat', 'latitude', 'lon', 'longitude', 'site']
```

Comment:
Is there a reason for this ? If this is meant to put the  coordinates at the top, maybe you could parse eventual `coordinates` attributes + 1D variables with the same name as their dimension.

Reply:
I think this is it. I pulled the code from the xarray implementation of cf-json, and this is what they did. 
https://github.com/metocean/cf-json/blob/master/cfjson/xrdataset.py